### DISCUSSION TITLE
ARXML checker/validator?

### DISCUSSION THREAD

**Discussion by nithinvelupur** [cogu/autosar#110] [Q&A] [answered]
Is there a module which can check the ARXML. Like a checker/validator? I would like to create AUTOSAR DEXT ARXMLs and would like to ensure the files are according to the DEXT specifications.

**Comment by cogu** [comment, marked as answer]
Personally, I validate XML using the Linux tool xmllint which is part of libxml2-utils. In Windows I run it either inside WSL2 or cygwin. Both ways works fine.

Here's the bash-command I use for validating all arxml files in current directory. Adapt the file name of the XSD to match the one you have.

```bash
find . -name "*.arxml" | xargs xmllint --noout --schema AUTOSAR_00051.xsd
```

You can download the AUTOSAR xml schemas freely from [autosar.org](https://autosar.org)
